src/kagglehub/gcs_upload.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+from datetime import datetime
+
+import requests
+
+from kagglehub.clients import KaggleApiV1Client
+from kagglehub.exceptions import postprocess_response
+
+logger = logging.getLogger(__name__)
+
+MAX_NUM_FILES = 50
+
+
+def parse(string: str):
+    time_formats = ["%Y-%m-%dT%H:%M:%S", "%Y-%m-%dT%H:%M:%SZ", "%Y-%m-%dT%H:%M:%S.%f", "%Y-%m-%dT%H:%M:%S.%fZ"]
+    for t in time_formats:
+        try:
+            result = datetime.strptime(string[:26], t).replace(microsecond=0)  # noqa: DTZ007
+            return result
+        except:  # noqa: E722, S110
+            pass
+    return string
+
+
+class File(object):  # noqa: UP004
+    def __init__(self, init_dict):
+        parsed_dict = {k: parse(v) for k, v in init_dict.items()}
+        self.__dict__.update(parsed_dict)
+        self.size = File.get_size(self.totalBytes)
+
+    def __repr__(self):
+        return self.ref
+
+    @staticmethod
+    def get_size(size, precision=0):
+        suffixes = ["B", "KB", "MB", "GB", "TB"]
+        suffix_index = 0
+        while size >= 1024 and suffix_index < 4:  # noqa: PLR2004
+            suffix_index += 1
+            size /= 1024.0
+        return "%.*f%s" % (precision, size, suffixes[suffix_index])
+
+
+def _upload_blob(file_path: str, model_type: str):
+    """Uploads a file to a remote server as a blob and returns an upload token.
+
+    Parameters
+    ==========
+    file_path: The path to the file to be uploaded.
+    model_type : The type of the model associated with the file.
+    """
+    data = {
+        "type": model_type,
+        "name": os.path.basename(file_path),
+        "contentLength": os.path.getsize(file_path),
+        "lastModifiedEpochSeconds": int(os.path.getmtime(file_path)),
+    }
+    api_client = KaggleApiV1Client()
+    response = api_client.post("/blobs/upload", data=data)
+    postprocess_response(response)
+
+    with open(file_path, "rb") as f:
+        file_data = f.read()
+
+    headers = {"Content-Type": "application/octet-stream"}
+    # TODO(aminmohamed): add resumable upload
+    requests.put(response["createUrl"], data=file_data, headers=headers, timeout=600, stream=True)
```

Comment:
I cant find it. seems like it gets passed in through this variable: `_request_timeout`. we can just set timeout to a high value like a day, maybe?

Reply:
Looks we don't pass a timeout when calling this GCS API: https://github.com/Kaggle/kaggle-api/blob/75dfa27d3c2886ea46e4d0c6bfcc0cc322af554b/kaggle/api/kaggle_api_extended.py#L3743C32-L3743C43

Looking at the documentation for the `requests` package:

> By default, requests do not have a timeout

Let's not specify it given we expect this operation to take a long time. It would be silly to cancel if it is still making progress.